### PR TITLE
Tools: Recommend what to do when astyle fails

### PR DESCRIPTION
**Pull request by Ryanf55**
# Purpose

Tell developers what to do if the astyle check fails. 

# Motivation

https://github.com/ArduPilot/ardupilot_wiki/pull/6292#discussion_r1777897717

# Demo

This is what a developer will see if they push code and CI fails. 

Specifically, we went from:
```
****** Files needing formatting found.
```
to:
```
****** Files needing formatting found. Please run ./Tools/scripts/run_astyle.py
```

In the screenshot below, I intentionally broke formatting in AP_DDS_Client.h. This is what the script will display in CI.
image

### CODE REVIEW

File: Tools/scripts/run_astyle.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def check(self):
             self.progress("astyle check failed: (%s)" % (ret.stdout))
             self.retcode = 1
         if "Formatted" in ret.stdout:
-            self.progress("Files needing formatting found")
+            self.progress("Files needing formatting found. Please run ./Tools/scripts/run_astyle.py")
```
Comment (peterbarker) [inline Tools/scripts/run_astyle.py:63]: This is a really strange thing to be printing in `run_astyle.py`!

If this is intended to make CI clearer, grepping the output of the dry-run might be a better idea.

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: Tools/scripts/run_astyle.py (hunk 1 above)
Comment (Ryanf55) [reply Tools/scripts/run_astyle.py:63]: Is your expectation that we only print this in `build_ci.sh`?

Comment (peterbarker) [reply Tools/scripts/run_astyle.py:63]: Yes, that's my expectation - don't confuse the user who is running this manually.

Comment (Ryanf55) [reply Tools/scripts/run_astyle.py]: Please look again. I've moved this message to build-ci.sh and check the error code.

**Review by peterbarker — APPROVED** [review]
LGTM

**Pull request merged**